js/script.js

```diff
@@ -74,142 +76,128 @@ function addBook(){
-            row.style.display = '';
-        } else {
-            row.style.display = 'none';
-        }
-    }
-}
-
-document.getElementById('costFilterMin').addEventListener('input', filterByCost);
-document.getElementById('costFilterMax').addEventListener('input', filterByCost);
-
-function filterByYear() {
-    const selectedYear = document.getElementById('yearDropdownFilter').value;
-    const table = document.querySelector('.table-container__table');
-    const rows = table.getElementsByTagName('tbody')[0].getElementsByTagName('tr');
-
-    for (const row of rows) {
-        const yearCell = row.cells[2];
-        const year = yearCell.textContent;
-
-        if (selectedYear === '' || year === selectedYear) {
-            row.style.display = '';
-        } else {
-            row.style.display = 'none';
-        }
-    }
-}
-
-document.getElementById('yearDropdownFilter').addEventListener('change', filterByYear);
-
-function restoreTable(){
-    const table = document.querySelector('.table-container__table');
-    const rows = table.getElementsByTagName('tbody')[0].getElementsByTagName('tr');
-
-    for (const row of rows) {
-        row.style.display = '';
-    }
-}
-
-function resetSorting(){
+function resetFiltering(){
     document.getElementById('idFilter').value = '';
     document.getElementById('titleFilter').value = '';
     document.getElementById('yearDropdownFilter').value = '';
     document.getElementById('authorFilter').value = '';
     document.getElementById('costFilterMin').value = '';
     document.getElementById('costFilterMax').value = '';
     document.getElementById('copiesFilter').value = '';
-    restoreTable();
+    refreshTable();
 }
 
-document.getElementById('removeIdFilter').addEventListener('click', function(){
+const filters = {
+    id: '',
+    title: '',
+    year: '',
+    author: '',
+    costMin: null,
+    costMax: null,
+    copies: '',
+};
+
+document.getElementById('idFilter').addEventListener('input', (e) => {
+    filters.id = e.target.value.toLowerCase();
+    applyFilters();
+});
+document.getElementById('titleFilter').addEventListener('input', (e) => {
+    filters.title = e.target.value.toLowerCase();
+    applyFilters();
+});
+document.getElementById('yearDropdownFilter').addEventListener('change', (e) => {
+    filters.year = e.target.value;
+    applyFilters();
+});
+document.getElementById('authorFilter').addEventListener('input', (e) => {
+    filters.author = e.target.value.toLowerCase();
+    applyFilters();
+});
+document.getElementById('costFilterMin').addEventListener('input', (e) => {
+    filters.costMin = parseFloat(e.target.value) || null;
+    applyFilters();
+});
+document.getElementById('costFilterMax').addEventListener('input', (e) => {
+    filters.costMax = parseFloat(e.target.value) || null;
+    applyFilters();
+});
+document.getElementById('copiesFilter').addEventListener('input', (e) => {
+    filters.copies = e.target.value.toLowerCase();
+    applyFilters();
+});

```

Comment:
it seems that this code is applied immediately after page load. Why it might be needed if initially all the filters have default values and shouldn't filter any row ?